### PR TITLE
Add dataset generation code

### PR DESCRIPTION
**Pull request by maxrjones**
Add notebooks to convert downscaled CMIP6 ensemble median to Zarr and generate pyramids with various chunk sizes

### CODE REVIEW

File: stores/utils.py [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+    ds_path : UPath
+        Path to input dataset
+    target : str
+        Path to write output data pyamid to
+    levels : int, optional
+        Number of levels in pyramid, by default 2
+
+    Returns
+    -------
+    target : str
```
Comment (maxrjones) [inline stores/utils.py]: The cmip6-downscaling functions had too much caching-specific code and not enough flexibility around chunks to be used directly, but we could consider putting these functions in carbonplan-data if they may be useful for future work

File: stores/utils.py [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+    dta = _pyramid_postprocess(dta, levels=levels, chunks=chunks, pixels_per_tile=pixels_per_tile)
+
+    # write to target
+    for child in dta.children.values():
+        for variable in child.ds.data_vars:
+            child[variable].encoding['write_empty_chunks'] = True
+
+    dta.to_zarr(target, mode='w')
+    validate_zarr_store(target)
+    return target
```
Comment (andersy005) [inline stores/utils.py]: As far as I know, most of this feature already exists in ndpyramid available at https://github.com/carbonplan/ndpyramid/blob/main/ndpyramid/utils.py. As a result, ndpyramid applies these steps here: https://github.com/carbonplan/ndpyramid/blob/c9b94f5f26455e3f361b8e8b3f1a3b6a61fbb231/ndpyramid/regrid.py#L285

Comment (maxrjones) [reply stores/utils.py]: Thanks for pointing this out! Fixed!

**Comment by maxrjones** [comment]
@andersy005 mind if I merge this PR? It shouldn't affect any other components.

**Comment by andersy005** [comment]
> @andersy005 mind if I merge this PR? It shouldn't affect any other components.

please go ahead and merge it when you get a moment (sorry i hadn't seen this comment until now)

**Pull request merged**